File: src/main/scala/com/rookie/submit/udf/SplitFunction.scala
package com.rookie.submit.udf

import org.apache.flink.table.annotation.DataTypeHint
import org.apache.flink.table.annotation.FunctionHint
import org.apache.flink.table.functions.TableFunction
import org.apache.flink.types.Row

@FunctionHint(output = new DataTypeHint("ROW<word STRING, length INT>"))
class SplitFunction extends TableFunction[Row] {

  def eval(str: String): Unit = {
    // use collect(...) to emit a row
    str.split(" ").foreach(s => collect(Row.of(s, Int.box(s.length))))
  }
}

File: src/main/java/com/rookie/submit/udf/SplitFunction.java
package com.rookie.submit.udf;

import org.apache.flink.table.annotation.DataTypeHint;
import org.apache.flink.table.annotation.FunctionHint;
import org.apache.flink.table.functions.TableFunction;
import org.apache.flink.types.Row;

@FunctionHint(output = @DataTypeHint("ROW<word STRING, length INT>"))
public class SplitFunction extends TableFunction<Row> {

    public void eval(String str) {
        if (str == null || str.isEmpty()) {
            return;
        }
        for (String s : str.split(" ")) {
            collect(Row.of(s, Integer.valueOf(s.length())));
        }
    }
}
